### PR TITLE
feat(Account): account constructor should take a KeyPair or a Signer …

### PR DESCRIPTION
**Pull request by ghost**
…as parameter

### CODE REVIEW

File: src/account/default.ts [hunk 1]
```diff
@@ -30,9 +30,10 @@ export class Account extends Provider implements AccountInterface {
 
   private signer: SignerInterface;
 
-  constructor(provider: Provider, address: string, keyPair: KeyPair) {
+  constructor(provider: Provider, address: string, keyPairOrSigner: KeyPair | Signer) {
```
Comment (janek26) [inline src/account/default.ts:33]: should be SignerInterface

File: src/account/default.ts [hunk 2]
```diff
@@ -30,9 +30,10 @@ export class Account extends Provider implements AccountInterface {
...
     super(provider);
-    this.signer = new Signer(keyPair);
+    this.signer =
+      keyPairOrSigner instanceof Signer ? keyPairOrSigner : new Signer(keyPairOrSigner as KeyPair);
```
Comment (janek26) [inline src/account/default.ts:36]: I have bad experiences with `instanceof` not detecting an instance.. Could we change this to something more low level?

**Review by janek26 — CHANGES_REQUESTED** [review]

File: src/account/default.ts (hunk 1 above)
Comment (jcaporossi-ledger) [reply src/account/default.ts]: yes fixed

File: src/account/default.ts (hunk 2 above)
Comment (jcaporossi-ledger) [reply src/account/default.ts]: okay, used "in" typeguard operator in the next commit....

**Review by janek26 — APPROVED** [review]

**Review by MilGard91 — APPROVED** [review]

**Pull request merged**